docs/pages/api-reference/liveblocks-react-comments.mdx

```diff
@@ -0,0 +1,852 @@
+their ID. To display user information based on IDs, you can provide a resolver
+function to the `resolveUser` option in
+[`createRoomContext`](#createRoomContext).
+
+This resolver function will receive a user ID and should return a user object
+asynchronously.
+
+```tsx file="liveblocks.config.ts"
+// ...
+
+export const {
+  /* ... */
+} = createRoomContext(client, {
+  resolveUser: (userId) => __getUserInfoFromDB__(userId),
+});
+```
+
+Once that’s done, components will show user names and avatars accordingly.
+
+#### Resolving mention suggestions [#createRoomContext.resolveMentionSuggestions]
+
+Liveblocks only knows users’ IDs, so to enable creating mentions in composer
+editors, you can provide a resolver function to the `resolveMentionSuggestions`
+option in [`createRoomContext`](#createRoomContext).
+
+This resolver function will receive the mention currently being written (e.g.
+when writing “@jane”, `text` will be `jane`) and should return a list of user
+IDs matching that text. This function will be called every time the text changes
+but with some debouncing.
+
+```tsx file="liveblocks.config.ts"
+// ...
+
+export const {
+  /* ... */
+} = createRoomContext(client, {
+  resolveMentionSuggestions: (text) => __getSuggestedUserIds__(text),
+});
+```
+
+### RoomProvider
+
+Makes comments from a room available in the component hierarchy below.
+
+```tsx
+import { RoomProvider } from "./liveblocks.config";
+
+function Panel() {
+  return <RoomProvider id="my-room">{/* children */}</RoomProvider>;
+}
+```
+
+## Types
+
+### ThreadData
+
+Represents a thread of comments.
+
+```tsx
+type ThreadData<ThreadMetadata> = {
+  type: "thread";
+  id: string;
+  roomId: string;
+  createdAt: string;
+  updatedAt?: string;
+  comments: CommentData[];
+  metadata: ThreadMetadata;
+};
+```
+
+### CommentData
+
+Represents a comment.
+
+```tsx
+type CommentData = {
+  type: "comment";
+  id: string;
+  threadId: string;
+  roomId: string;
+  userId: string;
+  body?: CommentBody;
+  createdAt: string;
+  editedAt?: string;
+  deletedAt?: string;
+};
+```
+
+## Hooks
+
+### useThreads
+
+Returns the threads within the current room.
+
+```tsx
+import { useThreads } from "./liveblocks.config";
+
+const { threads, error, isLoading } = useThreads();
+```
+
+### useUser
+
+Returns user info from a given user ID.
+
+```tsx
+import { useUser } from "./liveblocks.config";
+
+const { user, error, isLoading } = useUser("user-id");
+```
+
+<Banner title="Providing a resolver function">
+
+To use `useUser`, you should provide a resolver function to the
+[`resolveUser`](#createRoomContext.resolveUser) option in
+[`createRoomContext`](#createRoomContext).
+
+</Banner>
+
+### useCreateThread
+
+Returns a function that creates a thread with an initial comment, and optionally
+some metadata.
+
+```tsx
+const createThread = useCreateThread();
+const thread = createThread({ body: {}, metadata: {} });
+```
+
+### useEditThread
```

Comment:
Can threads be deleted too? No hook for this is mentioned. What if every comment in a thread is deleted?

Reply:
They can't be manually deleted no, only comments can. If every comment in a thread is deleted, the thread is automatically deleted.

Reply:
> They can't be manually deleted no, only comments can. If every comment in a thread is deleted, the thread is automatically deleted.

@marcbouchenoire Right! Could you make a note of that under `ThreadData`?

Reply:
I added a small note under `ThreadData`.

Happy to iterate more tomorrow.